File: code_examples/3-Structured_Problem_Solving/4-Scope_of_Variables/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module shows how local and global variables work
"""

# global variable
variable_x = 25


def local_variable() -> None:
    """The local_variable() function creates local variables, returns None."""
    variable_x = 10
    variable_x = variable_x + 1
    variable_y = 30
    variable_z = variable_x + variable_y
    print(f"Local variable:  {variable_x} + {variable_y} = {variable_z}")


def global_variable() -> None:
    """The global_variable() function uses a global variable, returns None."""
    global variable_x
    variable_x = variable_x + 1
    variable_y = 30
    variable_z = variable_x + variable_y
    print(f"Global variable: {variable_x} + {variable_y} = {variable_z}")


def main() -> None:
    """The main() function shows how local and global variables work, returns None."""
    local_variable()
    global_variable()

    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/3-Structured_Problem_Solving/4-Scope_of_Variables/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program shows how local and global variables work
 */

package main

import "fmt"

// global variable
var variableX = 25

func localVariable() {
	// this shows what happens with local variables
	var variableX = 10
	var variableY = 30

	variableX++
	variableZ := variableX + variableY

	fmt.Println("Local variableX, variableY, variableZ:", variableX, "+", variableY, "=", variableZ)
}

func globalVariable() {
	// this shows what happens with global variables
	variableY := 30

	variableX++
	variableZ := variableX + variableY

	fmt.Println("Global variableX, variableY, variableZ:", variableX, "+", variableY, "=", variableZ)
}

func main() {
	localVariable()
	globalVariable()

	fmt.Println("\nDone.")
}
